client/src/components/Dialog/InterestQuizDialog/QuizDialog.js

```diff
@@ -0,0 +1,135 @@
+    }
+});
+
+class QuizDialog extends React.Component {
+    constructor (props) {
+        super(props);
+        this.state = {
+            open: true,
+            selected: []
+        };
+
+        this.handleChange = this.handleChange.bind(this);
+        this.handleClose = this.handleClose.bind(this);
+        this.handleConfirm = this.handleConfirm.bind(this);
+    }
+
+    handleChange(interest) {
+        let selected = this.state.selected;
+        let index = selected.indexOf(interest);
+
+        if (index !== -1) {
+            selected.splice(index, 1);
+        } else if(selected.length <= 6) {
+            selected.push(interest);
+        }
+        this.setState({selected});
+    }
+
+    handleConfirm() {
+        console.log(this.state.selected);
+        this.setState({'open': false});
+    }
+
+    handleClose = () => {
+        this.setState({'open': false});
+    };
+
+    DialogTitle = withStyles(styles)(props => {
+        const { children, classes, onClose } = props;
+        return (
+            <MuiDialogTitle disableTypography className={classes.root}>
+                <Typography variant="h5" className={classes.title}>{children}</Typography>
+                {onClose ? (
+                    <IconButton aria-label="close" className={classes.closeButton} onClick={onClose} href={''}>
+                        <CloseIcon />
+                    </IconButton>
+                ) : null}
+            </MuiDialogTitle>
+        );
+    });
+
+    DialogContent = withStyles(theme => ({
+        root: {
+            paddingLeft: theme.spacing(8),
+            paddingRight: theme.spacing(8),
+            paddingTop: theme.spacing(2),
+            paddingBottom: theme.spacing(2),
+        },
+    }))(MuiDialogContent);
+
+    DialogActions = withStyles(theme => ({
+        root: {
+            margin: 0,
+            padding: theme.spacing(1),
+            marginBottom: '2rem'
+        },
+    }))(MuiDialogActions);
+
+    render() {
+        const { classes } = this.props;
+
+        return (
+            <Dialog onClose={this.handleClose} aria-labelledby="customized-dialog-title" open={this.state.open} maxWidth={'md'}>
+                <this.DialogTitle id="customized-dialog-title" onClose={this.handleClose}>
+                    Select your interests:
+                </this.DialogTitle>
+                <this.DialogContent>
+                    <Interests handleChange={this.handleChange} selected={this.state.selected} />
+                </this.DialogContent>
+                <this.DialogActions>
+                    <Button onClick={this.handleConfirm} color="primary" className={classes.button} href={''}>
+                        Done!
+                    </Button>
+                </this.DialogActions>
+            </Dialog>
+        );
+    }
+}
+
+const mapStateToProps = state => ({
+    userStore: state.UserStore
+});
+
+function mapDispatchToProps(dispatch) {
+    return bindActionCreators(
+        {
```

Comment:
is this an empty object?

Reply:
for now, add function later for integration